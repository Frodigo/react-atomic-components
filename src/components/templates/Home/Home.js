import React from 'react';

import PageHeader from '../../organisms/PageHeader';
import Hero from '../../molecues/Hero'
import Tile from '../../molecues/Tile';
import MediaBlock from '../../molecues/MediaBlock';
import ContentBlock from '../../molecues/ContentBlock';
import Heading from '../../atoms/Text/Heading';
import Link from '../../atoms/Link';
import Tout from '../../molecues/Tout';

import classes from './Home.module.css';
import FooterMock from "../partials/FooterMock";
import AboutTheAuthor from "../../organisms/AboutTheAuthor";

const Home = (props) => {
    const { ...rest } = props;

    const aboutTheAuthorProps = {
        heading: 'About me',
        name: 'John',
        image: 'https://via.placeholder.com/270x270',
        description:
            'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.',
        ctaUrl: '#',
        ctaLabel: 'Read more'
    };

    const pageHeaderNavItems = [
        {
            url: '#',
            label: 'Home',
            isActive: true
        },
        {
            url: '#',
            label: 'About me'
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch'
        }
    ];

    return (
        <div className={classes.Home} {...rest}>
            <PageHeader navItems={pageHeaderNavItems} logoUrl="https://via.placeholder.com/100x30" logoAlt="lorem ipsum" />
            <Hero heading="Lorem ipsum" image={{alt: 'figure', filename: 'https://via.placeholder.com/400x400'}} paragraph="Lorem ipsum dolor sit amet"/>
            <Tile heading="What I do?" />
            <div className={classes.mediaBlockList}>
                <MediaBlock
                    imageUrl="https://via.placeholder.com/290x240"
                    linkUrl="#"
                    text="Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. "
                    title="Lorem Ipsum"
                    imageWidth="290"
                    imageHeight="240"
                />
                <MediaBlock
                    imageUrl="https://via.placeholder.com/290x240"
                    linkUrl="#"
                    text="Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. "
                    title="Lorem Ipsum"
                    imageWidth="290"
                    imageHeight="240"
                />
            </div>
            <div className={classes.callToActionSection}>
                <ContentBlock
                    heading="Heading level 1"
                    paragraph="Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie"
                    cta={[{
                        href: '#',
                        children: 'Call to action',
                        variant: 'primary'
                    }]}
                />
            </div>
            <div className={classes.articles}>
                <header className={classes.articlesHeader}>
                    <Heading type="h4">Recent articles</Heading>
                </header>
                <div className={classes.articlesInner}>
                    <Tout
                        imageUrl="https://via.placeholder.com/290x240"
                        linkUrl="#"
                        title="Lorem ipsum"
                        imageWidth="290"
                        imageHeight="240"
                    />
                    <Tout
                        imageUrl="https://via.placeholder.com/290x240"
                        linkUrl="#"
                        title="Lorem ipsum"
                        imageWidth="290"
                        imageHeight="240"
                    />
                </div>
                <footer className={classes.articlesFooter}>
                    <Link href="#" variant="secondary">
                        Show all
                    </Link>
                </footer>
            </div>
            <AboutTheAuthor {...aboutTheAuthorProps} />
            <FooterMock/>
        </div>
    );
};

export default Home;
