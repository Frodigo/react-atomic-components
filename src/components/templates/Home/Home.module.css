.hero {
    position: relative;
    padding: var(--spacing-xl) var(--spacing-md);
}

.hero span {
    display: block;
}

.figure,
.figureDesktop {
    bottom: 15px;
    position: absolute;
    right: 15px;
}

.figureDesktop {
    display: none;
}

.marked {
    color: var(--color-secondary);
}

.callToActionSection {
    background-color: #d8d8d8;
}

.articles {
    padding: var(--spacing-xxl) 0 0 ;
}


.articlesHeader,
.articlesFooter {
    padding: 0 var(--spacing-md);
    margin-bottom: 40px;
    text-align: center;
}

.articlesFooter {
    margin-top: 40px;
}

@media (min-width: 768px) {
    .figure {
        display: none;
    }

    .figureDesktop {
        display: block;
        transform: scale(0.5);
        transform-origin: 100% 100%;
    }

    .mediaBlockList {
        padding-bottom: var(--spacing-xxl);
    }

    .articlesInner {
        display: flex;
    }

    .articlesInner > div {
        flex: 1 1 50%;
        margin: 0 var(--spacing-md);
    }
}

.followMe {
    border-top: 2px solid var(--color-text);
    background-color:#d8d8d8;

}

@media (min-width: 1024px) {
    .figureDesktop {
        transform: scale(0.8);
    }
}

@media (min-width: 1300px) {
    .hero,
    .mediaBlockList,
    .articles {
        margin: 0 auto;
        max-width: var(--desktop-container-width);
        padding: 60px 0;
    }

    .figureDesktop {
        transform: scale(1);
    }
}