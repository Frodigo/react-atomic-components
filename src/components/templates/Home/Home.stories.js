import React from 'react';

import { default as HomeComponent } from './Home';

export default {
    title: 'Templates',
    component: HomeComponent
};

const Template = (args) => <HomeComponent {...args} />;

export const Home = Template.bind({});
