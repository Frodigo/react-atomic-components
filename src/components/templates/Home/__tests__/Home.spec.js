import React from 'react';
import { render } from '@testing-library/react';

import Home from '../Home';

describe('Home page should be rendered', () => {
    test('Home page renders correctly', () => {
        const { container } = render(<Home />);

        expect(container).toMatchSnapshot();
    });
});
