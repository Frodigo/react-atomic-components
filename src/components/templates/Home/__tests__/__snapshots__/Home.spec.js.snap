// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should be rendered Home page renders correctly 1`] = `
<div>
  <div
    class="Home"
  >
    <header
      class="pageHeader"
    >
      <div
        class="container"
      >
        <a
          class="link"
          href="/"
        >
          <img
            alt="lorem ipsum"
            src="https://via.placeholder.com/100x30"
          />
        </a>
        <nav
          class="primaryNav"
        >
          <a
            class="navLinkActive navLink"
            href="#"
          >
            Home
          </a>
          <a
            class=" navLink"
            href="#"
          >
            About me
          </a>
          <a
            class=" navLink"
            href="#"
          >
            Blog
          </a>
          <a
            class=" navLink"
            href="#"
          >
            Get in touch
          </a>
        </nav>
      </div>
    </header>
    <div
      class="hero"
    >
      <figure
        class="figure"
      >
        <img
          alt="figure"
          class="image"
          src="https://via.placeholder.com/400x400"
        />
      </figure>
      <figure
        class="figureDesktop"
      >
        <img
          alt="figure"
          class="image"
          src="https://via.placeholder.com/400x400"
        />
      </figure>
      <section
        class="section"
      >
        <h1>
          Lorem ipsum
        </h1>
        <p
          style="text-transform: uppercase;"
        >
          Lorem ipsum dolor sit amet
        </p>
      </section>
    </div>
    <section
      class="tile"
    >
      <h2
        class="heading"
      >
        What I do?
      </h2>
    </section>
    <div
      class="mediaBlockList"
    >
      <div
        class="mediaBlock"
      >
        <div
          class="mediaBlockInner"
        >
          <figure
            class="image"
          >
            <img
              alt="Lorem Ipsum"
              height="240"
              src="https://via.placeholder.com/290x240"
              width="290"
            />
          </figure>
          <section
            class="content"
          >
            <h5
              class=""
            >
              Lorem Ipsum
            </h5>
            <p
              class="paraghaph"
            >
              Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. 
            </p>
          </section>
        </div>
      </div>
      <div
        class="mediaBlock"
      >
        <div
          class="mediaBlockInner"
        >
          <figure
            class="image"
          >
            <img
              alt="Lorem Ipsum"
              height="240"
              src="https://via.placeholder.com/290x240"
              width="290"
            />
          </figure>
          <section
            class="content"
          >
            <h5
              class=""
            >
              Lorem Ipsum
            </h5>
            <p
              class="paraghaph"
            >
              Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. 
            </p>
          </section>
        </div>
      </div>
    </div>
    <div
      class="callToActionSection"
    >
      <section
        class="block"
        style="background-color: transparent;"
      >
        <div
          class="container"
        >
          <h3
            class="heading"
          >
            Heading level 1
          </h3>
          <p
            class="paragraph"
          >
            Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie
          </p>
          <div
            class="links"
          >
            <a
              class="primary"
              href="#"
            >
              Call to action
            </a>
          </div>
        </div>
      </section>
    </div>
    <div
      class="articles"
    >
      <header
        class="articlesHeader"
      >
        <h4
          class=""
        >
          Recent articles
        </h4>
      </header>
      <div
        class="articlesInner"
      >
        <div
          class="tout"
        >
          <a
            href="#"
          >
            <figure
              class="figure"
            >
              <img
                alt="figure"
                height="240"
                layout="responsive"
                src="https://via.placeholder.com/290x240"
                width="290"
              />
            </figure>
            <header
              class="tape"
            >
              <h5
                class="inverse"
              >
                Lorem ipsum
              </h5>
            </header>
          </a>
        </div>
        <div
          class="tout"
        >
          <a
            href="#"
          >
            <figure
              class="figure"
            >
              <img
                alt="figure"
                height="240"
                layout="responsive"
                src="https://via.placeholder.com/290x240"
                width="290"
              />
            </figure>
            <header
              class="tape"
            >
              <h5
                class="inverse"
              >
                Lorem ipsum
              </h5>
            </header>
          </a>
        </div>
      </div>
      <footer
        class="articlesFooter"
      >
        <a
          class="secondary"
          href="#"
        >
          Show all
        </a>
      </footer>
    </div>
    <div
      class="aboutTheAuthor"
    >
      <div
        class="inner"
      >
        <figure
          class="figureDesktop"
          style="background-image: url(https://via.placeholder.com/270x270);"
        />
        <article
          class="content"
        >
          <h6
            class=""
          >
            About me
          </h6>
          <p
            class="authorName"
          >
            John
          </p>
          <figure
            class="figure"
            style="background-image: url(https://via.placeholder.com/270x270);"
          />
          <p
            class="description"
          >
            Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.
          </p>
          <a
            class="secondary"
            href="#"
          >
            Read more
          </a>
        </article>
      </div>
    </div>
    <form
      action="https://app.getresponse.com/add_subscriber.html"
      class="newsletter"
      id="newsletter"
      method="post"
    >
      <div
        class="newsletterContent"
      >
        <h3
          class="title"
        >
          Subscribe to the newsletter
        </h3>
        <p
          class="paraghaph"
        >
          I will send you unique content every Friday.
        </p>
        <div
          class="fieldset"
        >
          <div
            class="field"
          >
            <input
              class="input null"
              data-testid="input"
              name="first_name"
              placeholder="Your name"
              type="text"
              value=""
            />
          </div>
          <div
            class="field"
          >
            <input
              class="input null"
              data-testid="input"
              name="email"
              placeholder="Your email"
              type="email"
              value=""
            />
          </div>
        </div>
        <input
          name="campaign_token"
          type="hidden"
          value="WMdq9"
        />
        <input
          name="thankyou_url"
          type="hidden"
          value="https://marcin-kwiatkowski.com/thank-you"
        />
        <input
          name="start_day"
          type="hidden"
          value="0"
        />
        <footer
          class="footer"
        >
          <p>
            Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers
          </p>
          <button
            class="primary"
            type="submit"
          >
            Subscribe
          </button>
        </footer>
      </div>
    </form>
    <footer
      class="pageFooter"
    >
      <div
        class="columns"
      >
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 1
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 2
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 3
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 4
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
      </div>
      <div>
        <ul
          class="bottomNav"
        >
          <li>
            <a
              href="#"
            >
              Link 1
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Link 2
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Link 3
            </a>
          </li>
        </ul>
        <p
          class="copyright"
        >
          © 2021 Lorem ipsum
        </p>
      </div>
    </footer>
    <nav
      class="mobileNav"
    >
      <a
        href="/"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Home"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Home
          </span>
        </span>
      </a>
      <a
        href="/blog"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Blog"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Blog
          </span>
        </span>
      </a>
      <a
        href="/about-me"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="About"
            src="[object Object]"
          />
          <span
            class="label"
          >
            About
          </span>
        </span>
      </a>
      <a
        href="/get-in-touch"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Contact"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Contact
          </span>
        </span>
      </a>
      <a
        href="#newsletter"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Subscribe"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Subscribe
          </span>
        </span>
      </a>
    </nav>
  </div>
</div>
`;
